File: word/src/main/java/com/dreampany/word/data/model/Definition.java
package com.dreampany.word.data.model;

import android.arch.persistence.room.Ignore;
import android.os.Parcel;

import com.dreampany.frame.data.model.BaseParcel;
import com.google.common.base.Objects;

import org.jetbrains.annotations.NotNull;

/**
 * Created by Hawladar Roman on 2/9/18.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 */
public class Definition extends BaseParcel {

    private String partOfSpeech;
    private String text;

    public Definition() {
    }

    @Ignore
    private Definition(Parcel in) {
        super(in);
        partOfSpeech = in.readString();
        text = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
        dest.writeString(partOfSpeech);
        dest.writeString(text);
    }

    public static final Creator<Definition> CREATOR = new Creator<Definition>() {
        @Override
        public Definition createFromParcel(Parcel in) {
            return new Definition(in);
        }

        @Override
        public Definition[] newArray(int size) {
            return new Definition[size];
        }
    };

    @Override
    public boolean equals(Object in) {
        if (this == in) return true;
        if (in == null || getClass() != in.getClass()) return false;

        Definition item = (Definition) in;
        return Objects.equal(partOfSpeech, item.partOfSpeech) && Objects.equal(text, item.text);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(partOfSpeech, text);
    }

    public void setPartOfSpeech(@NotNull String partOfSpeech) {
        this.partOfSpeech = partOfSpeech;
    }

    public void setText(@NotNull String text) {
        this.text = text;
    }

    @NotNull
    public String getPartOfSpeech() {
        return partOfSpeech;
    }

    @NotNull
    public String getText() {
        return text;
    }
}

File: word/src/main/kotlin/com/dreampany/word/data/model/Definition.kt
package com.dreampany.word.data.model

import android.os.Parcel
import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Ignore
import com.dreampany.frame.data.model.BaseParcelKt
import com.dreampany.word.misc.Constants
import com.google.common.base.Objects
import com.google.firebase.firestore.PropertyName

/**
 * Created by Roman-372 on 7/22/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
data class Definition(
    @ColumnInfo(name = Constants.Word.PART_OF_SPEECH)
    @PropertyName(Constants.Word.PART_OF_SPEECH)
    var partOfSpeech: String?,
    var text: String?
) : BaseParcelKt() {

    constructor() : this("", "") {
    }

    @Ignore
    private constructor(parcel: Parcel) : this(parcel.readString(), parcel.readString()) {
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(partOfSpeech)
        dest.writeString(text)
    }

    companion object CREATOR : Parcelable.Creator<Definition> {
        override fun createFromParcel(parcel: Parcel): Definition {
            return Definition(parcel)
        }

        override fun newArray(size: Int): Array<Definition?> {
            return arrayOfNulls(size)
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val item = other as Definition
        return Objects.equal(item.partOfSpeech, partOfSpeech) && Objects.equal(item.text, text)
    }

    override fun hashCode(): Int {
        return Objects.hashCode(partOfSpeech, text)
    }


}